### PR TITLE
make graph_selection tests just checking selection

### PR DESCRIPTION
**Pull request by ChenyuLInx**
### Description
Fix graph tests since `expect_pass` is enabled for `run_dbt`. This change makes most of the tests in graph selection tests only check whether the selection is correct but not check whether the model run was successful. This makes the tests run faster(since we no longer need to do seed). In the future, if selection logic is moved out we could also just make this into unit tests.

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have added information about my change to be included in the [CHANGELOG](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry).

### CODE REVIEW

**Review by stu-k — COMMENTED** [review]
Nothing stands out, will leave for those more familiar with pytest and graph selection to approve 👍

**Review by gshank — APPROVED** [review]
This makes a lot of sense. All of the checking that tables actually exist really belongs to different functionality, not selection. And it was making these tests really slow.

There is a "check_result_nodes_by_name' which might be helpful for some of these tests. It uses "set" because sometimes the results come out in a different order. The order of the results isn't guaranteed.

**Comment by ChenyuLInx** [comment]
> This makes a lot of sense. All of the checking that tables actually exist really belongs to different functionality, not selection. And it was making these tests really slow.
> 
> There is a "check_result_nodes_by_name' which might be helpful for some of these tests. It uses "set" because sometimes the results come out in a different order. The order of the results isn't guaranteed.

Ha, I actually did all of them using set. But I will update it using `check_result_nodes_by_name` to promote implemented function.

File: tests/functional/graph_selection/test_graph_selection.py
Comment (jtcohen6) [inline tests/functional/graph_selection/test_graph_selection.py:43]: Do we want this to just be `list`, rather than `run` + expect fail?

(feel free to ignore me and ship as is)

Comment (ChenyuLInx) [reply tests/functional/graph_selection/test_graph_selection.py:43]: That's a great idea! updating

Comment (ChenyuLInx) [reply tests/functional/graph_selection/test_graph_selection.py:43]: Actually `ls` and `run` doesn't seems to select the exact same items, it would include source, exposure etc. It would be much bigger change including change the actual expected results to make the update. I think we can leave it as an item if we plan to touch more things related to selector

Comment (jtcohen6) [reply tests/functional/graph_selection/test_graph_selection.py:43]: Oh good call. You can limit this to models (same as `run`) with `dbt ls --resource-type model`. But very happy with leaving it as is

Comment (ChenyuLInx) [reply tests/functional/graph_selection/test_graph_selection.py:43]: Ops, this poped up after I hit the merge button. This is good to know!

**Pull request merged**
